.github/workflows/test.yml

```diff
@@ -5,18 +5,47 @@ on:
     branches:
       - main
       - dev
-    paths-ignore:
-      - "docs/**"
-      - "scripts/**"
-      - "**/README.md"
   pull_request:
-    paths-ignore:
-      - "docs/**"
-      - "**/*.md"
 
 jobs:
+  changes:
+    runs-on: ubuntu-latest
+    outputs:
+      lint: ${{ steps.filter.outputs.lint }}
+      lint_files: ${{ steps.filter.outputs.lint_files }}
+    steps:
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          list-files: shell
+          filters: |
+            lint:
+              - added|modified: "docs/**"
+              - added|modified: "scripts/**"
+              - added|modified: "**/*.md"
+
+  filter:
+    needs: [changes]
+    runs-on: ubuntu-latest
+    outputs:
+      DOCS_ONLY: ${{ steps.docs.outputs.DOCS_ONLY }}
+    steps:
+      - id: docs
+        run: |
+          for file in ${{ needs.changes.outputs.lint_files }}; do
+            echo $file
+            echo ".."
+          done
+          echo ::set-output name=DOCS_ONLY::true
```

Comment:
This is tricky haha.  I wonder if it would be easier if we just had a separate `lint` workflow running on all pull requests and kept the test workflow skipping docs-only changes?

Reply:
i mostly just don't want 100 workflows 😅